solidity/core/contracts/validator-manager/MultisigValidatorManager.sol

```diff
@@ -0,0 +1,239 @@
+        _unenrollValidator(_validator);
+    }
+
+    /**
+     * @notice Sets the quorum threshold.
+     * @param _quorumThreshold The new quorum threshold.
+     */
+    function setQuorumThreshold(uint256 _quorumThreshold) external onlyOwner {
+        _setQuorumThreshold(_quorumThreshold);
+    }
+
+    /**
+     * @notice Gets the addresses of the current validator set.
+     * @dev There ordering guarantee due to the semantics of EnumerableSet.AddressSet.
+     * @return The addresses of the validator set.
+     */
+    function validatorSet() external view returns (address[] memory) {
+        uint256 _length = validators.length();
+        address[] memory _validatorSet = new address[](_length);
+        for (uint256 i = 0; i < _length; i++) {
+            _validatorSet[i] = validators.at(i);
+        }
+        return _validatorSet;
+    }
+
+    // ============ Public Functions ============
+
+    /**
+     * @notice Returns whether provided signatures over a checkpoint constitute
+     * a quorum of validator signatures.
+     * @dev Reverts if _signatures is not sorted in ascending order by the signer
+     * address, which is required for duplicate detection.
+     * @dev Does not revert if a signature's signer is not in the validator set.
+     * @param _root The merkle root of the checkpoint.
+     * @param _index The index of the checkpoint.
+     * @param _signatures Signatures over the checkpoint to be checked for a validator
+     * quorum. Must be sorted in ascending order by signer address.
+     * @return TRUE iff _signatures constitute a quorum of validator signatures over
+     * the checkpoint.
+     */
+    function isQuorum(
+        bytes32 _root,
+        uint256 _index,
+        bytes[] calldata _signatures
+    ) public view returns (bool) {
+        uint256 _signaturesLength = _signatures.length;
+        // If there are less signatures provided than the required quorum threshold,
+        // this is not a quorum.
+        if (_signaturesLength < quorumThreshold) {
+            return false;
+        }
+        // To identify duplicates, the signers recovered from _signatures
+        // must be sorted in ascending order. previousSigner is used to
+        // enforce ordering.
+        address _previousSigner = address(0);
+        uint256 _validatorSignatureCount = 0;
+        for (uint256 i = 0; i < _signaturesLength; i++) {
+            address _signer = _recoverCheckpointSigner(
+                _root,
+                _index,
+                _signatures[i]
+            );
+            // Revert if the signer violates the required sort order.
+            require(_previousSigner < _signer, "!sorted signers");
+            // If the signer is a validator, increment _validatorSignatureCount.
+            if (validators.contains(_signer)) {
+                _validatorSignatureCount++;
+            }
+            _previousSigner = _signer;
+        }
+        return _validatorSignatureCount >= quorumThreshold;
```

Comment:
I intentionally don't short circuit the loop once `_validatorSignatureCount == quorumThreshold` because I assume the agent will be smart enough to only submit the number of signatures necessary to reach quorum and adding an additional check in the loop would add gas costs. Thoughts?